### ISSUE TITLE
feature request: consider add Ticker

### ISSUE THREAD

**Issue by Sherlock-Holo** [async-rs/futures-timer#60]
consider adding a new struct `Ticker`, which implement `futures::Stream`, can return an instant periodically by use `.next().await`

**Comment by antifuchs** [comment]
In the absence of this, I quickly wrote & released https://crates.io/crates/futures-ticker ([crate](https://crates.io/crates/futures-ticker)) - happy to merge it into this crate, or keep it independent (and implement it for different Timer schemes, e.g. smol's).

**Comment by yoshuawuyts** [comment]
Thanks for opening @Sherlock-Holo! I guess we may want to re-introduce [`async_std::stream::interval`](https://docs.rs/async-std/1.6.2/async_std/stream/fn.interval.html) into this crate.

**Comment by kennetpostigo** [comment]
@yoshuawuyts Hi, i'm looking for this functionality exactly, being able to run some work at every interval

**Comment by bittrance** [comment]
Indeed, I ended up at this issue because I tried to figure out what happened to Interval from the legacy futures version of futures-timer.

**Comment by youngjoon-lee** [comment]
Hi, Is there any plan to merge https://github.com/async-rs/futures-timer/issues/60#issuecomment-650601845 into this repo?
